src/database/models.py

```diff
@@ -29,41 +32,41 @@ def __str__(self):
         return self.organization_name
 
 
-class User(models.Model):
-    user_email = models.EmailField(primary_key=True)
-    first_name = models.CharField(max_length=100)
-    last_name = models.CharField(max_length=100)
+class UserProfile(models.Model):
+    user = models.OneToOneField(User, on_delete=models.CASCADE, primary_key=True)
     phone_number = models.CharField(max_length=20, blank=True, null=True)
     active = models.BooleanField(default=True)
     created_at = models.DateTimeField(auto_now_add=True)
     modified_at = models.DateTimeField(auto_now=True)
 
     def __str__(self):
-        return f"{self.first_name} {self.last_name}"
+        return self.user.username
+
+
+@receiver(post_save, sender=User)
+def create_or_update_user_profile(sender, instance, created, **kwargs):
+    if created:
+        UserProfile.objects.create(user=instance)
+    else:
+        instance.userprofile.save()
 
 
 class OrganizationAdmin(models.Model):
-    admin_id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-    organization_id = models.ForeignKey(
-        Organization, on_delete=models.CASCADE, to_field="organization_id"
-    )
-    user_email = models.ForeignKey(
-        User, on_delete=models.CASCADE, to_field="user_email"
-    )
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    organization = models.ForeignKey(Organization, on_delete=models.CASCADE)
+    user = models.ForeignKey(User, on_delete=models.SET_NULL, null=True, blank=True)
     access_level = models.CharField(max_length=50)
     active = models.BooleanField(default=True)
     created_at = models.DateTimeField(auto_now_add=True)
     modified_at = models.DateTimeField(auto_now=True)
 
     def __str__(self):
-        return f"{self.user_email.first_name} {self.user_email.last_name} - {self.organization_id.organization_name}"
+        return f"{self.user.get_full_name() if self.user else 'No User'} - {self.organization.name}"
```

Comment:
I think this should be {self.organization.organization_name} because name is not a field in organization

Reply:
will solve now